forage-android/src/main/java/com/joinforage/forage/android/core/services/forageapi/network/ForageApiResponse.kt

```diff
@@ -97,101 +95,17 @@ sealed class ForageApiResponse<out T> {
      * handle the error and display the appropriate
      * [customer-facing message](https://docs.joinforage.app/docs/document-error-messages#template-error-table-for-fns-documentation).
      */
-    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>() {
-        internal companion object {
-            fun fromError(error: ForageError): Failure {
-                return Failure(listOf(error))
-            }
-        }
-    }
-}
```

Comment:
A huge part of this PR is replacing this companion object pattern with constructors since the later is less verbose and less complicated to maintain